src/sagemaker/sklearn/processing.py

```diff
@@ -83,7 +84,16 @@ def __init__(
         if not command:
             command = ["python3"]
 
-        session = sagemaker_session or Session()
+        if instance_type in ("local", "local_gpu"):
+            session = sagemaker_session or LocalSession()
+            if not isinstance(session, LocalSession):
+                raise RuntimeError(
+                    "instance_type local or local_gpu is only supported with an"
+                    "instance of LocalSession"
+                )
+        else:
+            session = sagemaker_session or Session()
```

Comment:
very similar, maybe identical block of code in `sagemaker.processing` module. may want to consider if abstraction is possible to eliminate repeated code, especially if it's used a few times.

Reply:
Implemented on base class only. Please check.